exchange/utils.go

```diff
@@ -227,21 +233,35 @@ func getAuctionBidderRequests(req AuctionRequest,
 		}
 	}
 
-	reqExt, err := getExtJson(req.BidRequest, requestExt)
-	if err != nil {
-		return nil, []error{err}
-	}
-
 	var errs []error
 	for bidder, imps := range impsByBidder {
 		coreBidder := resolveBidder(bidder, aliases)
 
 		reqCopy := *req.BidRequest
 		reqCopy.Imp = imps
-		reqCopy.Ext = reqExt
 
 		prepareSource(&reqCopy, bidder, sChainsByBidder)
 
+		if len(bidderParamsInReqExt) != 0 {
+
+			params, err := getBidderParamsForBidder(bidderParamsInReqExt, bidder)
+			if err != nil {
+				return nil, []error{err}
+			}
+
+			if params != nil && requestExt == nil {
+				requestExt = &openrtb_ext.ExtRequest{}
+			}
+			requestExt.Prebid.BidderParams = params
```

Comment:
This will cause an error if `params` is nil and `requestExt` is nil. If both are nil, the above if fails, but we should not have to do anything.

Reply:
Removed confusing check for if params != nil && requestExt == nil. It is not required as we already have check if len(bidderParamsInReqExt) != 0 at line number 245. Code will not enter if block at line 245 if requestExt is nil.